src/Engine/ProtoCore/Reflection/MemberInfoReflectionExtensions.cs

```diff
@@ -0,0 +1,211 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using System.Text;
+using System.Threading.Tasks;
+using Autodesk.DesignScript.Runtime;
+
+namespace ProtoCore.Reflection
+{
+    internal static class MemberInfoReflectionExtensions
+    {
+        internal static MethodInfo GetBaseDefinitionReflectionContext(this MethodInfo methodInfo)
+        {
+            try
+            {
+                // if overriding object 
+                var baseType = methodInfo.DeclaringType.BaseType;
+                if (baseType is null)
+                {
+                    return methodInfo;
+                }
+
+                // This will fail if there are overloads of the method
+                var baseTypeMethod = baseType.GetMethod(methodInfo.Name);
```

Comment:
If the expected behaviour with overloads is to return `null`, would it be better to use a `Where` filter query and check the methods with that name is 1.

If > 1, return null.